### PR TITLE
Add firewall documentation for the Azure Logs integration

### PR DESCRIPTION
**Pull request by zmoog**
## Proposed commit message

Specify the TCP ports required to enable Azure Logs on an Agent behind a firewall.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] ~~I have verified that all data streams collect metrics or logs.~~
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] ~~I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).~~

## Related issues

- Relates #9157

### CODE REVIEW

**Review by gizas — APPROVED** [review]

**Review by alaudazzi — APPROVED** [review]
I left a few editing suggestions, otherwise LGTM.

**Comment by willemdh** [comment]
@zmoog The only remark I have here is that 5672 is unencrypted AMQP traffic and should not be opened. I can confirm it is not needed, it works when only 5671 is opened. I'd really advice to check

```
*.servicebus.windows.net:5671
*.blob.core.windows.net:443

Also when you nslookup the servicebus URL, you get 3 *.cloudapp.net Url's which Also need to be allowed.
```

When the `*.cloudapp.net` url's are not opened, it does not work

**Comment by zmoog** [comment]
> The only remark I have here is that 5672 is unencrypted AMQP traffic and should not be opened. I can confirm it is not needed, it works when only 5671 is allowed. I'd really advice to check

> When the *.cloudapp.net url's are not allowed, it does not work

Thank you @willemdh, I'll double check both the ports and the URLs.

**Comment by zmoog** [comment]
According to the [AMQP 1.0 in Azure Service Bus and Event Hubs protocol guide](https://learn.microsoft.com/en-us/azure/service-bus-messaging/service-bus-amqp-protocol-guide#amqp-outbound-port-requirements), is seems the service on port 5672 performs a mandatory upgrade of connection to TLS:

> Azure Service Bus or Azure Event Hubs requires the use of TLS at all times. It supports connections over TCP port 5671, whereby the TCP connection is first overlaid with TLS before entering the AMQP protocol handshake, and also supports connections over TCP port 5672 whereby the server immediately offers a mandatory upgrade of connection to TLS using the AMQP-prescribed model. 

In the same document, in the [AMQP outbound port requirements](https://learn.microsoft.com/en-us/azure/service-bus-messaging/service-bus-amqp-protocol-guide#amqp-outbound-port-requirements) section, it lists both ports:

> Clients that use AMQP connections over TCP require ports 5671 and 5672 to be opened in the local firewall.

So my current understanding is that using both 5671 and 5672 ports is not insecure because the service requires a mandatory upgrade to TLS. 

I also noticed the document mentions the need for additional ports the EnableLinkRedirect is enabled:

> Along with these ports, it might be necessary to open additional ports if the [EnableLinkRedirect](https://learn.microsoft.com/en-us/dotnet/api/microsoft.servicebus.messaging.amqp.amqptransportsettings.enablelinkredirect) feature is enabled. EnableLinkRedirect is a new messaging feature that helps skip one-hop while receiving messages, thus helping to boost throughput. The client would start communicating directly with the back-end service over port range 104XX [...]

Personally, I never enable this option, but I want to learn more.

**Comment by zmoog** [comment]
@alaudazzi, I made a few minor additions. Would you mind taking a final look? :bow:

File: packages/azure/_dev/build/docs/README.md [hunk 1]
```diff
@@ -266,6 +266,59 @@ The Elastic Agent can use one Storage account container for all integrations.
 
 The Agent will use the integration name and the event hub name to identify the blob to store the consumer group information uniquely.
 
+### Running the integration behind a firewall
+
+When running the Elastic Agent behind a firewall, you need to allow specific traffic to ensure proper communication with the necessary components. In this case, you need to allow traffic on port `5671` and `5672` for the Event Hub, and port `443` for the Storage Account container.
```
Comment (alaudazzi) [inline packages/azure/_dev/build/docs/README.md:271]:
```suggestion
When you run the Elastic Agent behind a firewall, to ensure proper communication with the necessary components, you need to allow traffic on port `5671` and `5672` for the Event Hub, and port `443` for the Storage Account container.
```

File: packages/azure/_dev/build/docs/README.md [hunk 2]
```diff
@@ -266,6 +266,59 @@ The Elastic Agent can use one Storage account container for all integrations.
...
+└─Azure──────────────────────────┘
+```
+
+#### Event Hub
+
+Port `5671` and `5672` are commonly used for secure communication with the Event Hub. These ports are used to receive events. By allowing traffic on these ports, the Elastic Agent can establish a secure connection with the Event Hub. 
+
+For more information, check the following documents:
+
+- The section [What ports do I need to open on the firewall?](https://learn.microsoft.com/en-us/azure/event-hubs/event-hubs-faq#what-ports-do-i-need-to-open-on-the-firewall) from the [Event Hubs frequently asked questions](https://learn.microsoft.com/en-us/azure/event-hubs/event-hubs-faq#what-ports-do-i-need-to-open-on-the-firewall).
```
Comment (alaudazzi) [inline packages/azure/_dev/build/docs/README.md:305]:
```suggestion
- [What ports do I need to open on the firewall?](https://learn.microsoft.com/en-us/azure/event-hubs/event-hubs-faq#what-ports-do-i-need-to-open-on-the-firewall) from the [Event Hubs frequently asked questions](https://learn.microsoft.com/en-us/azure/event-hubs/event-hubs-faq#what-ports-do-i-need-to-open-on-the-firewall).
```

File: packages/azure/_dev/build/docs/README.md [hunk 3]
```diff
@@ -266,6 +266,59 @@ The Elastic Agent can use one Storage account container for all integrations.
...
+- [AMQP outbound port requirements](https://learn.microsoft.com/en-us/azure/service-bus-messaging/service-bus-amqp-protocol-guide#amqp-outbound-port-requirements)
+
+#### Storage Account Container
+
+Port `443` is used for secure communication with the Storage Account container. This port is commonly used for HTTPS traffic. By allowing traffic on port 443, the Elastic Agent can securely access and interact with the Storage Account container, which is essential for storing and retrieving checkpoint data for each event hub partition.
+
+#### DNS
+
+Optionally, if you want to restrict traffic to domain names, here is the list of names you need to allow.
```
Comment (alaudazzi) [inline packages/azure/_dev/build/docs/README.md:314]:
```suggestion
Optionally, you can restrict the traffic to the following domain names:
```

**Review by alaudazzi — APPROVED** [review]
A few editing suggestions. LGTM.

**Pull request merged**